src/WorldIDIdentityManagerImplV3.sol

```diff
@@ -0,0 +1,233 @@
+    /// @param params parameters for the process defined by the `RegisterIdentities4844Params` structure.
+    /// @custom:reverts Unauthorized If the message sender is not authorised to add identities.
+    /// @custom:reverts NotLatestRoot If the provided `params.preRoot` is not the latest root.
+    /// @custom:reverts ProofValidationFailure If `params.insertionProof` cannot be verified using the
+    ///                 provided inputs.
+    /// @custom:reverts VerifierLookupTable.NoSuchVerifier If the batch sizes doesn't match a known
+    ///                 verifier.
+    function registerIdentities(
+        RegisterIdentities4844Params calldata params
+    ) public virtual onlyProxy onlyInitialized onlyIdentityOperator {
+        if (params.preRoot != _latestRoot) {
+            revert NotLatestRoot(params.preRoot, _latestRoot);
+        }
+
+        // No matter what, the inputs can result in a hash that is not an element of the scalar
+        // field in which we're operating. We reduce it into the field before handing it to the
+        // verifier. All other elements that are passed as calldata are reduced in the circuit.
+        uint256 reducedElement = uint256(params.inputHash) % SNARK_SCALAR_FIELD;
+        uint256 evaluationReduced = params.expectedEvaluation % SNARK_SCALAR_FIELD;
+
+        // We need to look up the correct verifier before we can verify.
+        ITreeVerifier insertionVerifier =
+                            batchInsertion4844Verifiers.getVerifierFor(params.batchSize);
+
+        // TODO are we using always this one slot or should it be configurable or calculated?
+        bytes32 kzgCommitmentHash = blobhash(0);
+
+        // With that, we can properly try and verify.
+        try insertionVerifier.verifyProof(
+            params.insertionProof,
+            params.commitments,
+            params.commitmentPok,
+            [reducedElement,
+            evaluationReduced,
+            uint256(kzgCommitmentHash),
+            uint256(params.startIndex),
+            params.preRoot,
+            params.postRoot]
+        ) {
+            // If it did verify, we need to update the contract's state. We set the currently valid
+            // root to the root after the insertions.
+            _latestRoot = params.postRoot;
+
+            // We also need to add the previous root to the history, and set the timestamp at
+            // which it was expired.
+            rootHistory[params.preRoot] = uint128(block.timestamp);
+
+            bool success = evaluatePoint(
+                kzgCommitmentHash,
+                bytes32(params.kzgChallenge),
+                bytes32(params.expectedEvaluation),
+                params.kzgCommitment,
+                params.kzgProof
+            );
+            require(success, "Call to point evaluation precompiled contract failed");
+
+            emit TreeChanged(params.preRoot, TreeChange.Insertion, params.postRoot);
+        } catch Error(string memory errString) {
```

Comment:
What's the point of the try/catch block here? Seems like we always revert so why not revert at the top of the function scope?

Reply:
Ok, I see it's consistent with V1 and V2. Let's keep it like this